scripts/download_technique_animations.py

```diff
@@ -8,67 +8,126 @@
-    cleaned_filename = os.path.splitext(filename)[0]
+    cleaned_filename = os.path.splitext(os.path.basename(filepath))[0]
     cleaned_filename = cleaned_filename.strip("0123456789_")
     cleaned_filename = cleaned_filename.lower()
     return cleaned_filename
 
 
-def process_gif(filename: str) -> None:
+def process_animation_name(filename: str) -> str:
+    """Convert filename of format 'file_name' to a capitalized 'File Name' animation name/title."""
+    animation_name = " ".join([name_part.capitalize() for name_part in filename.split("_")])
+    return animation_name
+
+
+def download_animation_credits(gif_page_url: str) -> str:
+    """Extract credits text from a GIF animation subpage, from the 'Comments' section."""
+    gif_filename: str = gif_page_url.split("File:")[-1]
+    animation_name: str = process_animation_name(process_filename(gif_filename))
+
+    gifpage_source = requests.get(gif_page_url)
+    gifpage_tree = html.fromstring(gifpage_source.content)
+    comment_blocks = gifpage_tree.xpath("//table[@class='wikitable filehistory']//td[@dir='ltr']")
+
+    credits_text =""
+    for comment_row in comment_blocks: 
+        # Some comment entries are empty
+        if len(comment_row) > 0:
+            for comment_entry in comment_row:
+                # Assemble credits and skip the "Category:" blocks   
+                comment_tail: str = comment_entry.tail
+                if comment_tail is not None and not comment_tail.startswith("Category:"):
+                    credits_text += comment_entry.tail.strip()
+
+    credits_text = credits_text or "Unknown animation source/author"
+    credits_record: str = CREDITS_TEMPLATE.format(animation_name=animation_name, animation_url=gif_page_url, credits_text=credits_text)
+    return credits_record
+
+
+def gif_to_frames(filepath: str) -> None:
     """Extract individual animation frames as PNG from a GIF."""
-    with Image.open(filename) as image:
+    with Image.open(filepath) as image:
         if not image.is_animated:
-            print(f"{filename} is not animated, skipped")
+            print(f"{filepath} is not animated, skipped")
             return
         if not image.width == image.height == FRAME_SIZE:
-            print(f"{filename} is not {FRAME_SIZE}x{FRAME_SIZE}, skipped")
+            print(f"{filepath} is not {FRAME_SIZE}x{FRAME_SIZE}, skipped")
             return
-        base_name = process_filename(filename)
+  
+        base_name: str = process_filename(filepath)
+        animation_name: str = process_animation_name(base_name)
         for frame in range(0, image.n_frames):
-            frame_filename = f"{base_name}{frame:02}.png"
-            print(f"saving {filename}:{frame_filename} {frame}/{image.n_frames - 1}")
+            frame_filename: str = os.path.join(ANIMATION_DIR, f"{base_name}{frame:02}.png")
+            print(f"Generating animation frames for '{animation_name}': {frame_filename} - {frame}/{image.n_frames - 1}")
```

Comment:
Gives us the following records:
```
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade00.png - 0/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade01.png - 1/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade02.png - 2/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade03.png - 3/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade04.png - 4/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade05.png - 5/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade06.png - 6/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade07.png - 7/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade08.png - 8/9
Generating animation frames for 'Fire Fade': /home/amender/projects/Tuxemon/mods/tuxemon/animations/technique/fire_fade09.png - 9/9
```
The absolute path is super verbose, but at least we see where the frame files end up on disk.